src/caffeine/program_termination_s.f90

```diff
@@ -50,40 +62,44 @@ end subroutine prif_stop_character
   end procedure prif_stop
 
   module procedure prif_error_stop
-
-    !TODO: deal with argument `quiet`
     if (present(stop_code_char)) then
-       call prif_error_stop_character(stop_code_char)
-    else if (present(stop_code_int)) then
-       call prif_error_stop_integer(stop_code_int)
+       call prif_error_stop_character(quiet, stop_code_char)
     else
-       call prif_error_stop_integer()
+       call prif_error_stop_integer(quiet, stop_code_int)
     end if
   end procedure prif_error_stop
 
-  subroutine prif_error_stop_character(stop_code)
+  subroutine prif_error_stop_character(quiet, stop_code)
     !! stop all images and provide the stop_code as the process exit status
+    logical(c_bool), intent(in) :: quiet
     character(len=*), intent(in) :: stop_code
 
-    write(error_unit, *) stop_code
-    flush error_unit
+    if (.not. quiet) then
+      write(error_unit, *) stop_code
```

Comment:
IIUC the details of what output formatting are unspecified by Fortran (and PRIF), but for symmetry with `prif_stop_character` and `prif_error_stop_integer` shouldn't this be something like:
```suggestion
      write(error_unit, *) "ERROR STOP '" // stop_code // "'"
```

Reply:
Thanks for the catch, I did want that change and symmetry, but I forgot it. Appreciate it!